### PR TITLE
decrease metadata response timeout duration

### PR DESCRIPTION
**Pull request by desniaak**

### CODE REVIEW

**Comment by desniaak** [comment]
> what do you think if we make those variables configurable when creating the client?

to be more flexible for the client yes, feedback has been addressed.

**Comment by thoriqsatriya** [comment]
> to be more flexible for the client yes, feedback has been addressed.

If we do that, it will break the current version right? I think we need to tag a version first and increase the version for this

**Comment by fadhillahentino** [comment]
you're last build was failed

WARN [runner/megacheck] Can't run megacheck because of compilation errors in packages [github.com/AccelByte/eventstream-go-sdk [github.com/AccelByte/eventstream-go-sdk.test]]: kafka_test.go:30: too few arguments in call to NewKafkaClient 
kafka_test.go:30:61: too few arguments in call to NewKafkaClient (typecheck)
	client, err := NewKafkaClient("test", []string{"localhost"})

**Comment by fadhillahentino** [comment]
> > to be more flexible for the client yes, feedback has been addressed.
> 
> If we do that, it will break the current version right? I think we need to tag a version first and increase the version for this

what do you think if we use variadic argument for the config? so, it would be optional and SDK have a default config?

**Comment by desniaak** [comment]
> > > to be more flexible for the client yes, feedback has been addressed.
> > 
> > 
> > If we do that, it will break the current version right? I think we need to tag a version first and increase the version for this
> 
> what do you think if we use variadic argument for the config? it would be optional and SDK have a default config
> So it would be like this
> func NewKafkaClient(realm string, brokerList []string, config ...*KafkaConfig)
> 
> with this approach, client is able to set the config or not and also not breaking even they increase the version.

I don't think so, because it is to distinguish configuration between sync and async producer

**Comment by desniaak** [comment]
> > > to be more flexible for the client yes, feedback has been addressed.
> > 
> > 
> > If we do that, it will break the current version right? I think we need to tag a version first and increase the version for this
> 
> what do you think if we use variadic argument for the config? it would be optional and SDK have a default config
> So it would be like this
> func NewKafkaClient(realm string, brokerList []string, config ...*KafkaConfig)
> 
> with this approach, client is able to set the config or not and also not breaking even they increase the version.

I don't think so, because it is to distinguish configuration between sync and async producer

File: kafka.go [hunk 1]
```diff
@@ -31,17 +31,40 @@ type KafkaClient struct {
...
+func NewKafkaClient(realm string, brokerList []string, producerConfiguration ...*KafkaConfig) (*KafkaClient, error) {
+
+	var configAsync, configSync *sarama.Config
+	switch {
+	case len(producerConfiguration) > 1:
+		configAsync = configureKafka(producerConfiguration[0])
+		configSync = configureKafka(producerConfiguration[1])
+	case len(producerConfiguration) == 1:
+		configAsync = configureKafka(producerConfiguration[0])
+		configSync = sarama.NewConfig()
```
Comment (fadhillahentino) [inline kafka.go]: Instead of just create a new config, I think we need a default value for KafkaConfig

Comment (desniaak) [reply kafka.go]: it is actually creating config with default value

Comment (fadhillahentino) [reply kafka.go]: I mean default value that we defined for event stream so we dont use default value from sarama

Comment (desniaak) [reply kafka.go]: the purpose to make these configurable is to replace default value from sarama so I don't think we need default value set in this library too.

Comment (fadhillahentino) [reply kafka.go]: oh i see, its okay for me

**Pull request merged**